goto/gotomagic/magic.py

```diff
@@ -61,6 +62,19 @@ def update_shortcut(self, magicword, uri):
             )
             exit(1)
 
+    def rename_shortcut(self, from_magicword, to_magicword, overwrite=False):
+        """ Renaming a shortcut """
+        if not overwrite and to_magicword in self.magic:
+            uri = self.magic[to_magicword]
+            return GotoWarning('adding_existing_magicword_short', magicword=to_magicword, uri=uri)  # noqa
+
+        if from_magicword not in self.magic:
+            return GotoWarning('magicword_does_not_exist', magicword=from_magicword)  # noqa
+
+        from_uri = self.magic[from_magicword]
```

Comment:
uri or url? Should we just go for one of them? I vote for url 👍 Goto is a locator tool. It gives you locations of stuff, based on uniform resource locators.

Reply:
We aee actually working with uris, and some of the uris we are working with are urls.

I actually would want developers going deep enough to contribute to goto, to be intrigued by the use of uri in sthe source code and ask themself, what is the difference between uris and urls.


In the help text however, I think it is more.. practical, to use urls over uris. to ease adoptation of new users. 

the file paths for instance, are not urls, but they are uris.

Reply:
Meanwhile
```
jonas@8600k:~$ project add uri-vs-url
active project is now: urivsurl

jonas@8600k:~$ goto list

jonas@8600k:~$ goto add venn-diagram-1 https://res.cloudinary.com/practicaldev/image/fetch/s--lrbx3qNQ--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://thepracticaldev.s3.amazonaws.com/i/j4bka41nypm4do1f3e5b.JPG
Added magic word venn-diagram-1

jonas@8600k:~$ goto add what-is-the-difference-between-a-uri-and-a-url https://dev.to/flippedcoding/what-is-the-difference-between-a-uri-and-a-url-4455
Added magic word what-is-the-difference-between-a-uri-and-a-url

jonas@8600k:~$ goto add venn-diagram-2 https://danielmiessler.com/images/uri-urn-url-miessler-2019.png
Added magic word venn-diagram-2

jonas@8600k:~$ goto add the-difference-between-urls-uris-and-urns https://danielmiessler.com/study/url-uri/
Added magic word the-difference-between-urls-uris-and-urns

jonas@8600k:~$ goto list
the-difference-between-urls-uris-and-urns
venn-diagram-1
venn-diagram-2
what-is-the-difference-between-a-uri-and-a-url

jonas@8600k:~$ 
```

Reply:
rfc3305#confusion - https://tools.ietf.org/html/rfc3305#section-2.3